orchagent/buffer/bufferhelper.cpp

```diff
@@ -0,0 +1,303 @@
+// includes -----------------------------------------------------------------------------------------------------------
+
+#include <unordered_map>
+#include <string>
+
+#include <tokenize.h>
+
+#include "bufferschema.h"
+
+#include "bufferhelper.h"
+
+using namespace swss;
+
+// helper -------------------------------------------------------------------------------------------------------------
+
+void BufferHelper::parseBufferConfig(BufferProfileConfig &cfg) const
+{
+    auto &map = cfg.fieldValueMap;
+
+    const auto &cit = map.find(BUFFER_PROFILE_PACKET_DISCARD_ACTION);
+    if (cit != map.cend())
+    {
+        cfg.isTrimmingEligible = cit->second == BUFFER_PROFILE_PACKET_DISCARD_ACTION_TRIM ? true : false;
+    }
+}
+
+void BufferHelper::parseBufferConfig(BufferPriorityGroupConfig &cfg) const
+{
+    auto &map = cfg.fieldValueMap;
+
+    const auto &cit = map.find(BUFFER_PG_PROFILE);
+    if (cit != map.cend())
+    {
+        cfg.profile.value = cit->second;
+        cfg.profile.is_set = true;
+    }
+}
+
+void BufferHelper::parseBufferConfig(IngressBufferProfileListConfig &cfg) const
+{
+    auto &map = cfg.fieldValueMap;
+
+    const auto &cit = map.find(BUFFER_PORT_INGRESS_PROFILE_LIST_PROFILE_LIST);
```

Comment:
Does bufferorch use ingress/egress buffer profile list for trim eligible validation?

Reply:
@kperumalbfn validation is currently implemented for the next cases:

1. Buffer Profile is attached to either Ingress PG or Ingress/Egress Buffer Profile List and user tries to set trimming eligibility
2. Buffer Profile is configured as trimming eligible and user tries to attach it either to Ingress PG or Ingress/Egress Buffer Profile list

In both cases the error will be generated.
This guarantees that trimming eligible Buffer Profile can be used only with the Queue objects.